### PR TITLE
🎯 Improve eval routing and add SAMPLE_RATE support

### PR DESCRIPTION
**Pull request by TechNickAI**
## Summary

- **Fix speed hint priority** in concierge routing - user saying "quick" now properly routes to Haiku even when content is complex
- **Add SAMPLE_RATE env var** for stratified sampling in evals - enables faster iteration

## Eval Improvements

**Routing Eval:**
| Metric | Before | After |
|--------|--------|-------|
| Model Selection | 90.91% | **100.00%** ✅ |
| Temperature | 100% | 100% |
| Tool Invocation | 100% | 100% |
| Reasoning | 100% | 100% |

**Competitive Eval:**
| Metric | Before | After |
|--------|--------|-------|
| HTTP Success | 96% | **100%** ✅ |
| Infrastructure Health | 96% | **100%** ✅ |
| Tool Used | 80% | **100%** ✅ |
| Response Substance | 96% | **100%** ✅ |
| Latency | 62% | 62% (expected) |

## Changes

1. **lib/concierge/prompt.ts**: Added example showing "quick" + complex content → Haiku, strengthened conflict resolution language to explicitly include model selection
2. **evals/routing/eval.ts**: Added SAMPLE_RATE env var with stratified sampling
3. **evals/competitive/eval.ts**: Added SAMPLE_RATE env var with stratified sampling

## Test plan

- [x] Run full routing eval - all metrics 100%
- [x] Run full competitive eval - all metrics 100% (except latency which is expected for complex queries)
- [x] All 1570 unit/integration tests pass

## Notes

Known issue identified but not fixed: Context overflow (200K+ tokens) fails completely rather than truncating. This requires implementing message truncation/summarization which is documented as future work in knowledge/components/context-window-management.md.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

----

> [!IMPORTANT]
> Improve eval routing by fixing speed hint priority and adding SAMPLE_RATE for stratified sampling, enhancing model selection and infrastructure health.
> 
>   - **Behavior**:
>     - Fix speed hint priority in `lib/concierge/prompt.ts` to ensure "quick" routes to Haiku even with complex content.
>     - Add `SAMPLE_RATE` env var for stratified sampling in `evals/routing/eval.ts` and `evals/competitive/eval.ts`.
>   - **Eval Improvements**:
>     - Routing Eval: Model Selection improved from 90.91% to 100%.
>     - Competitive Eval: HTTP Success, Infrastructure Health, Tool Used, and Response Substance improved to 100%.
>   - **Misc**:
>     - Add example in `lib/concierge/prompt.ts` for "quick" + complex content routing to Haiku.
>     - Strengthen conflict resolution language in `lib/concierge/prompt.ts` to include model selection.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=carmentacollective%2Fcarmenta&utm_source=github&utm_medium=referral)<sup> for 21e768e71f235848badbbcb96c03319545903780. You can [customize](https://app.ellipsis.dev/carmentacollective/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by TechNickAI** [comment]
Fixed in 21e768e:
- Removed non-functional shuffle (was producing identical hashes for all items in each category)
- Renamed unused `category` variable to `_category` 
- Updated comments to reflect actual behavior (sequential sampling maintains consistency)

Great catches on the shuffle logic - all three bots correctly identified that the hash function was producing identical values within each category group.

File: evals/routing/eval.ts [hunk 1]
```diff
@@ -253,19 +253,56 @@ async function executeMultiTurnTest(input: TestInput): Promise<RoutingOutput> {
...
+    }
+
+    console.log(
+        `📊 Sampling ${sampled.length}/${data.length} test cases (${(sampleRate * 100).toFixed(0)}%)`
+    );
+    return sampled;
+}
+
+// Configuration for sampling
+const SAMPLE_RATE = parseFloat(process.env.SAMPLE_RATE ?? "1.0");
```
[the remark this answers is not in the document]
Comment (TechNickAI) [reply evals/routing/eval.ts]: Declining - invalid env vars are a user error. The current behavior (NaN flows through and produces empty array via slice(0, NaN)) is acceptable for this edge case. If someone sets SAMPLE_RATE=abc, they'll quickly notice zero tests ran from the console output.

Adding || 1.0 fallback would mask the invalid input. Better to fail obviously than silently default when the user made a typo.
